lib/stemcell/metadata_source.rb

```diff
@@ -43,7 +43,7 @@ def default_options
       DEFAULT_OPTIONS.merge(config.default_options)
     end
 
-    def expand_role(role, environment, override_options={}, options={})
+    def expand_role(role, environment, override_contexts=[], override_options={}, options={})
```

Comment:
Nitpick: should we name this variable `contexts` instead of `override_contexts` (also in metadata_launcher.rb). The reason I suggest this is because `override_options` is similarly named (prefixed with `override_`) but the `override_` prefix in `override_options` means something different although the similarity implies they are related.